### PR TITLE
Update to use Firebase cloud messaging on Android channel

### PR DESCRIPTION
**Pull request by norkans7**

### CODE REVIEW

File: app/src/main/java/io/rapidpro/androidchannel/RapidProFirebaseMessageService.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
+package io.rapidpro.androidchannel;
+
+import com.google.firebase.messaging.FirebaseMessagingService;
+import com.google.firebase.messaging.RemoteMessage;
+
+import java.util.Map;
+
+
+public class RapidProFirebaseMessageService extends FirebaseMessagingService {
```
Comment (nicpottier) [inline app/src/main/java/io/rapidpro/androidchannel/RapidProFirebaseMessageService.java:9]: Why prefix these with RapidPro?

**Review by nicpottier — CHANGES_REQUESTED** [review]
Apart from naming looks good, what did we discover with permissions for SMS?

File: app/src/main/java/io/rapidpro/androidchannel/RapidProFirebaseMessageService.java (hunk 1 above)
Comment (norkans7) [reply app/src/main/java/io/rapidpro/androidchannel/RapidProFirebaseMessageService.java:9]: This is a subclass of FirebaseMessageService, any reason to not use RapidPro?

**Comment by norkans7** [comment]
first service intents should use the service class, and the provider need to have expoerted set to true and optionally but as best practice both need to specify a permission to limit who can act on them

File: app/src/main/java/io/rapidpro/androidchannel/RapidProFirebaseMessageService.java (hunk 1 above)
Comment (ericnewcomer) [reply app/src/main/java/io/rapidpro/androidchannel/RapidProFirebaseMessageService.java:9]: Temba?

... oh I guess we use RapidPro as the main app.

Comment (norkans7) [reply app/src/main/java/io/rapidpro/androidchannel/RapidProFirebaseMessageService.java:9]: yep, so I change all of them? or just this one only?

Comment (ericnewcomer) [reply app/src/main/java/io/rapidpro/androidchannel/RapidProFirebaseMessageService.java:9]: I'd disregard my comment, I don't see a strong reason to go renaming everything else.

Comment (norkans7) [reply app/src/main/java/io/rapidpro/androidchannel/RapidProFirebaseMessageService.java:9]: sawa

**Comment by norkans7** [comment]
@nicpottier @ericnewcomer @rowanseymour please review once again

File: packs/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="utf-8"?>
 <manifest xmlns:android="http://schemas.android.com/apk/res/android"
-          package="@CONFIG.APP_PACKAGE_NAME@"
```
Comment (ericnewcomer) [inline packs/src/main/AndroidManifest.xml:3]: Isn't this important for building the other branded versions?

File: packs/src/main/AndroidManifest.xml [hunk 3]
```diff
@@ -12,7 +12,7 @@
     <uses-permission android:name="android.permission.READ_SMS" />
     <uses-permission android:name="android.permission.WRITE_SMS" />
 
-    <permission android:name="io.rapidpro.androidchannel.SEND_MESSAGE" android:label="SMS Channel"></permission>
```
Comment (ericnewcomer) [inline packs/src/main/AndroidManifest.xml:15]: I think this may be intentionally ambiguous so as to not confused branded versions

**Review by ericnewcomer — CHANGES_REQUESTED** [review]
Have you made sure that you can build a branded version of this? I'm worried that some of the manifest changes and build config stuff here might have broken that.

File: packs/src/main/AndroidManifest.xml (hunk 2 above)
Comment (norkans7) [reply packs/src/main/AndroidManifest.xml:3]: I am using gradle to build now this will do the right thing using applicationID

Comment (norkans7) [reply packs/src/main/AndroidManifest.xml:3]: Ohh no I need one more step in the script for this

Comment (ericnewcomer) [reply packs/src/main/AndroidManifest.xml:3]: Right, but did you confirm you can build the TextIt branded version as well? We can't break that workflow.

File: packs/src/main/AndroidManifest.xml (hunk 3 above)
Comment (norkans7) [reply packs/src/main/AndroidManifest.xml:15]: this one is well handled RapidPro is replaced by Brand

File: packs/src/main/AndroidManifest.xml (hunk 2 above)
Comment (norkans7) [reply packs/src/main/AndroidManifest.xml:3]: Now I confirm I built a branded version too

**Review by ericnewcomer — APPROVED** [review]

**Pull request merged**